libcloud/compute/drivers/outscale.py

```diff
@@ -73,6 +76,23 @@ def __init__(self,
         }
 
     def list_locations(self, ex_dry_run: bool = False):
+        """
+        Lists available locations details.
+
+        :param      ex_dry_run: If true, checks whether you have the required
+        permissions to perform the action.
+        :type       ex_dry_run: ``bool``
+        :return: locations details
+        :rtype: ``dict``
+        """
+        action = "ReadLocations"
+        data = json.dumps({"DryRun": ex_dry_run})
+        response = self._call_api(action, data)
+        if response.status_code == 200:
```

Comment:
Eventually we should move some of this error handling code to the Response class - this way we can avoid all this duplicated lines, but that's fine for now.